test/KetlAttestation.deploy.ts

```diff
@@ -0,0 +1,185 @@
+import { BigNumber, providers } from 'ethers'
+import { DEV_KETL_ATTESTATION_CONTRACT } from '@big-whale-labs/constants'
+import { KetlAttestation, KetlAttestation__factory } from '../typechain'
+import { ethers, upgrades } from 'hardhat'
+import { expect } from 'chai'
+import { getFakeAttestationCheckerVerifier } from './utils/fakes'
+import {
+  getLegacyMintCalldata,
+  getLegacyRegisterEntanglementCalldata,
+  getLegacyTokenHolders,
+} from '../scripts/helpers'
+
+describe('KetlAttestation Deploy Tests', () => {
+  it('should be able to deploy KetlAttestation and populate data correctly', async function () {
+    const uri = 'https://game.example/api/item/{id}.json'
+    const version = '0.0.1'
+
+    const realProvider = new providers.JsonRpcProvider(
+      process.env.ETH_RPC,
+      'maticmum'
+    )
+    const realKetlAttesation = await KetlAttestation__factory.connect(
+      DEV_KETL_ATTESTATION_CONTRACT,
+      realProvider
+    )
+    const attestorPublicKey = await realKetlAttesation.attestorPublicKey()
+
+    this.signers = await ethers.getSigners()
+    this.owner = this.signers[0]
+    this.user = this.signers[1]
+
+    this.poseidonT3Factory = await ethers.getContractFactory('PoseidonT3')
+    this.poseidonT3 = await this.poseidonT3Factory.deploy()
+    await this.poseidonT3.deployed()
+
+    this.incrementalBinaryTreeFactory = await ethers.getContractFactory(
+      'IncrementalBinaryTree',
+      {
+        libraries: {
+          PoseidonT3: this.poseidonT3.address,
+        },
+      }
+    )
+    this.incrementalBinaryTree =
+      await this.incrementalBinaryTreeFactory.deploy()
+    await this.incrementalBinaryTree.deployed()
+
+    this.ketlAttestationFactory = await ethers.getContractFactory(
+      'KetlAttestation',
+      {
+        libraries: {
+          IncrementalBinaryTree: this.incrementalBinaryTree.address,
+        },
+      }
+    )
+
+    this.fakeAttestationCheckerVerifier =
+      await getFakeAttestationCheckerVerifier(this.owner)
+    this.fakePasswordCheckerVerifier = await getFakeAttestationCheckerVerifier(
+      this.owner
+    )
+    await this.fakeAttestationCheckerVerifier.mock.verifyProof.returns(true)
+
+    this.ketlAttestation = (await upgrades.deployProxy(
+      this.ketlAttestationFactory,
+      [
+        uri,
+        version,
+        attestorPublicKey,
+        this.fakeAttestationCheckerVerifier.address,
+        this.fakePasswordCheckerVerifier.address,
+        ethers.constants.AddressZero,
+      ],
+      { unsafeAllowLinkedLibraries: true }
+    )) as KetlAttestation
+    await this.ketlAttestation.deployed()
+
+    const attestationMerkleRoots: { [key: number]: BigNumber } = {}
+    for (const attestationType of [0, 1, 2, 3]) {
```

Comment:
And re-use enum here

Reply:
Fixed in https://github.com/BigWhaleLabs/ketl-attestation-token/pull/33/commits/c9f0df7480fc31a632cb7cc93667374caa4f9b6a.